src/ddmd/backend/cgelem.c

```diff
@@ -2280,6 +2278,70 @@ STATIC elem * eldiv(elem *e, goal_t goal)
 
     if (OPTIMIZER)
     {
+        elem *e1 = e->E1;
+        if (tyintegral(tym) && e->Eoper == OPdiv && e2->Eoper == OPconst &&
+            e1->Eoper == OPdiv && e1->E2->Eoper == OPconst)
+        {
+            /* Replace:
+             *   (e / c1) / c2
+             * With:
+             *   e / (c1 * c2)
+             */
+            targ_llong c1 = el_tolong(e1->E2);
+            targ_llong c2 = el_tolong(e2);
+            bool uns1 = tyuns(e1->E1->Ety) || tyuns(e1->E2->Ety);
+            bool uns2 = tyuns(e1->Ety) || tyuns(e2->Ety);
+            if (uns1 == uns2)   // identity doesn't hold for mixed sign case
+            {
+                // The transformation will fail if c1*c2 overflows. This substitutes
+                // for a proper overflow check.
+                if (uns1 ? (c1 < 0x10000000 && c2 < 0x10000000)
```

Comment:
I thought I'd leave proper overflow checking for the D conversion. In the meantime, I'd stick with the values being less than sqrt(int.max).